receipt_dynamo/receipt_dynamo/entities/entity_mixins.py

```
@@ -814,6 +814,69 @@ def calculate_centroid(self) -> Tuple[float, float]:
+            height: The height of the image to scale coordinates. Defaults to None.
+            flip_y: Whether to flip the y coordinate. Defaults to False.
+
+        Returns:
+            Tuple of four corner coordinates: (top_left, top_right, bottom_left, bottom_right)
+            Each corner is a (x, y) tuple.
+
+        Raises:
+            ValueError: If only one of width or height is provided.
+        """
+        if (width is None) != (height is None):
+            raise ValueError("Both width and height must be provided together")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)